utils.py

```diff
@@ -1,16 +1,91 @@
 """Utility functions."""
+import functools
+from pathlib import Path
+
+from flask import request
+from openapi_core import create_spec
+from openapi_core.contrib.flask import FlaskOpenAPIRequest
+from openapi_core.validation.request.validators import RequestValidator
+from openapi_spec_validator import validate_spec
+from openapi_spec_validator.readers import read_from_filename
+from werkzeug.exceptions import BadRequest, UnsupportedMediaType
+
+from kytos.core import log
 from kytos.core.events import KytosEvent
 
 
 def emit_event(controller, name, **kwargs):
     """Send an event when something happens with an EVC."""
-    event_name = f'kytos/mef_eline.{name}'
+    event_name = f"kytos/mef_eline.{name}"
     event = KytosEvent(name=event_name, content=kwargs)
     controller.buffers.app.put(event)
 
 
 def compare_endpoint_trace(endpoint, vlan, trace):
     """Compare and endpoint with a trace step."""
-    return endpoint.switch.dpid == trace['dpid']\
-        and endpoint.port_number == trace['port']\
-        and vlan == trace['vlan']
+    return (
+        endpoint.switch.dpid == trace["dpid"]
+        and endpoint.port_number == trace["port"]
+        and vlan == trace["vlan"]
+    )
+
+
+def load_spec():
+    """Validate openapi spec."""
+    napp_dir = Path(__file__).parent
+    yml_file = napp_dir / "openapi.yml"
+    spec_dict, _ = read_from_filename(yml_file)
+
+    validate_spec(spec_dict)
+
+    return create_spec(spec_dict)
+
+
+def validate(spec):
+    """Decorator to validate a REST endpoint input.
+
+    Uses the schema defined in the openapi.yml file
+    to validate.
+    """
+
+    def validate_decorator(func):
+        @functools.wraps(func)
+        def wrapper_validate(*args, **kwargs):
+            print(f"Spec {spec}")
+            try:
+                data = request.get_json()
+            except BadRequest:
+                result = "The request body is not a well-formed JSON."
+                log.debug("create_circuit result %s %s", result, 400)
+                raise BadRequest(result) from BadRequest
+            if data is None:
+                result = "The request body mimetype is not application/json."
+                log.debug("update result %s %s", result, 415)
+                raise UnsupportedMediaType(result)
+
+            validator = RequestValidator(spec)
+            openapi_request = FlaskOpenAPIRequest(request)
```

Comment:
@ajoaoff it's not directly related to this change here, but thinking that openapi-core will be a core part of our validations,
how are we planning to validate payloads that we have on `listen_to` metods that some NApps have? the same author of openapi-core has published this [openapi-schema-validator](https://github.com/p1c2u/openapi-schema-validator) library to validate an object against the schema without being coupled with the request.  However, it quickly tired it out and I hit some `jsonschema.exceptions.RefResolutionError` when trying to use the resulting spec from the `load_spec`, so looks like if we'll use this we might need to auto resolve the schema. How are we envisioning this part?

Reply:
I don't think it is something we will do now, but one option would be using `pydantic`.

Reply:
Yeah `pydantic` is great, if in the future we find a way to also integrate it seamlessly, looks like we could also one day to generate `pydantic` models given the spec https://pydantic-docs.helpmanual.io/datamodel_code_generator/, cool.